src/components/CommitmentDetails.tsx

```diff
@@ -173,17 +182,22 @@ const CommitmentDetails = (props: DetailProps): JSX.Element => {
       ) : null}
 
       {loading ? (
-        <p>Loading...</p>
+        <LoopIcon />
       ) : (
         <div>
-          <div
-            onClick={() => {
-              setDisplayId('');
-              window.location.hash = '';
-            }}
-          >
+          <div>
             <div className={classes.titleContainer}>
-              <h2 className={classes.pageTitle}>Recurring Payment</h2>
+              <div className={classes.buttonContainer}>
+                <Button variant="outlined" color="primary" size="small">
+                  <ArrowBackIcon
+                    onClick={() => {
+                      window.history.back();
+                      setDisplayId('');
```

Comment:
You should either do `window.history.back()` or `setDisplayId('')` not both.  There's a distinction here: using `window.history.back()` would take the user to the last page they were on.  This page could be anything, it does not necessary have to be the commitments table.  Like a user could have been on google.com then navigated directly to this page.  `window.history.back()` would take us to google.com, but `setDisplayId('')` would take us to the commitments table.